pkg/cacheutil/redis_client.go

```diff
@@ -232,7 +266,22 @@ func NewRedisClientWithConfig(logger log.Logger, name string, config RedisClient
 			config.MaxSetMultiConcurrency,
 			gate.Sets,
 		),
+		setAsyncCircuitBreaker: noopCircuitBreaker{},
+	}
+	if config.SetAsyncCircuitBreakerEnabled {
+		c.setAsyncCircuitBreaker = gobreakerCircuitBreaker{gobreaker.NewCircuitBreaker(gobreaker.Settings{
+			Name:        "redis-set-async",
+			MaxRequests: config.SetAsyncCircuitBreakerHalfOpenMaxRequests,
+			Interval:    10 * time.Second,
+			Timeout:     config.SetAsyncCircuitBreakerOpenDuration,
+			ReadyToTrip: func(counts gobreaker.Counts) bool {
```

Comment:
Should this be factored out into another function? I think it's exactly the same in `memcached_client`.

Reply:
I have separated the entire configuration and logic, including this function. Please take another look.